### PR TITLE
Club house project

### PR DESCRIPTION
**Pull request by Leon-Mbegera**
This is a clubhouse project where members or its users can write anonymous posts.
- Only if a member is signed in is when they can see the author of a post, otherwise, they will only see the title and context of the post.
- It includes links for new members to sign up, links to sign in, and another link to sign out.

### CODE REVIEW

File: app/views/devise/passwords/edit.html.erb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+<%= form_for(resource, as: resource_name, url: password_path(resource_name), html: { method: :put }) do |f| %>
+  <%= render "devise/shared/error_messages", resource: resource %>
+  <%= f.hidden_field :reset_password_token %>
+
+  <div class="field">
+    <%= f.label :password, "New password" %><br />
+    <% if @minimum_password_length %>
+      <em>(<%= @minimum_password_length %> characters minimum)</em><br />
+    <% end %>
```
Comment (SajjadAhmad14) [inline app/views/devise/passwords/edit.html.erb:11]: Please create a helper under app/helpers and move this logic there. [OPTIONAL]

**Review by SajjadAhmad14 — CHANGES_REQUESTED** [review]
# Changes Required :recycle:
> "Change is inevitable. Change is constant. -Benjamin Disraeli"
## Hello, you have been doing a tremendous job so far :ok_hand:
## To Highlight 🎉
- [x] README has all the instructions to run the project locally.
- [x] Commit messages are descriptive.
## To Improve 🙏
- [x] Please use proper Git flow. The main/master branch should be there which is used for production releases.
- [x] PR should have a descriptive summary.
- [x] User should be able to sign in. Please see this [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/sign_in_requirement.png).
- [x] A signed-in user you can create a new post. Please see this [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/signed_in_can_post_requirement.png)
- [x] A signed-in user can see the post's author. Please see this [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/signed_in_can_see_author_requirement.png).
- [x] User should be able to sign out. Please see this [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/sign_out_requirement.png).
- [x] A guest user can only see the post's text. Please see this [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/guest_can_see_only_posts_requirement.png).
- [x] Sign in/Sign out links should be present in any view of the app. Please see this [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/sign_in_out_link_requirement.png).
- [x] Posts controller should use #before_action to :new and :create. Please see this [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/before_action_requirement.png).
- [x] Project does not have Users controller. Please make sure it has a controller in order to authenticate the user.
- [x] App has no functionality as per requirements. The app shows only this page:
issues

- [x] Please see inline comments for reference.
# Optional
- [x] Views should be completely separate from app logic. You can use app/helpers to move all of your view-related logic.

### Happy Coding ☺️

## If you have any questions you can comment here in this PR or can always connect with me on slack ( @Sajjad Ahmad ).

**Comment by Leon-Mbegera** [comment]
I have addressed all the requested changes herein,
Thanks for the straightforward guidelines on what to change.
gracias`

**Review by george-shammar — CHANGES_REQUESTED** [review]
### **STATUS:CHANGES REQUIRED** :recycle: :recycle: :construction: :construction_worker_man: 

Decent effort on this project. :+1: 

**Highlight**

- [x] Thank you for using the correct gitflow and implementing some of the required changes.

Please see required changes below:

- [x] As a signed-in user, you can create a new post. There is no where on your page to create a new post without manually editing the url which is not ideal.
- [x] A signed-in user should be able to see the post's author. But if posts cannot be created, it can't be seen.
- [x] A guest user should only see the post's text. Same reason as above - post cannot be created.
- [x] Sign in/Sign out links should be present in any view of the app. There is no way to link other pages of the application. 
- [ ] Project does not have Users controller. In trying to implement this, you have created a huge "User" directory and commented out every code inside all the files. Please get rid of all commented unused codes.

Please make the required changes and submit another CRR when you're ready.

In need of clarification or support, please drop your message here and ping me on slack.

Cheers and happy coding. :+1: :smile:

**Comment by Leon-Mbegera** [comment]
Herein, I have addressed the requested changes:

1. As a signed user, he/she can create a post
2. Only a signed-in user can see the author of a post
3. A guest user can only see the post text
4. I've displayed sign up/sign in, and sign out links on every page. Possible to navigate to other pages
5. I've deleted all unused code in the user controller.

**Review by SajjadAhmad14 — APPROVED** [review]
# Status: Approved :heavy_check_mark: :heavy_check_mark: :heavy_check_mark:

## Congratulations 🎉 Thank you for implementing changes suggested previously 🙏
# OPTIONAL
- [x] You can show flash messages for success and failure for user-friendliness.
- [x] Views should be completely separate from app logic. Please move view-related logic under app/helpers.
### Happy Coding ☺️

## If you have any questions you can comment here in this PR or can always connect with me on slack ( @Sajjad Ahmad ).

**Pull request merged**